cranelift/src/bugpoint.rs

```diff
@@ -652,6 +710,22 @@ impl Mutator for MergeBlocks {
     }
 }
 
+fn const_for_type<'f, T: InstBuilder<'f>>(builder: T, ty: ir::Type) -> &'static str {
+    // Try to keep the result type consistent, and default to an integer type
+    // otherwise: this will cover all the cases for f32/f64 and integer types, or
+    // create verifier errors otherwise.
+    if ty == F32 {
+        builder.f32const(0.0);
+        "f32const"
+    } else if ty == F64 {
+        builder.f64const(0.0);
+        "f64const"
+    } else {
+        builder.iconst(ty, 0);
+        "iconst"
+    }
```

Comment:
Don't we need a `bconst` for the boolean types?

Reply:
I just moved this function from `ReplaceInstWithConst`, but yeah, we should add it.

Reply:
Ok: https://github.com/bytecodealliance/wasmtime/issues/1633